### PR TITLE
chore: add telemetry span around collecting garbage

### PR DESCRIPTION
**Pull request by mjhenkes**
- Closes 

### Additional details

Adds a span around collecting garbage so we can compare develop with v13

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/server/lib/browsers/memory/index.ts [hunk 1]
```diff
@@ -274,10 +275,15 @@ const checkMemoryPressureAndLog = async ({ automation, test }: { automation: Aut
 const checkMemoryPressure: (automation: Automation) => Promise<void> = measure(async (automation: Automation) => {
   if (collectGarbageOnNextTest) {
     debug('forcing garbage collection')
+    let span
+
     try {
+      telemetry.startSpan({ name: 'checkMemoryPressure:collect:garbage' })
```
Comment (AtofStryker) [inline packages/server/lib/browsers/memory/index.ts]:
```suggestion
      span = telemetry.startSpan({ name: 'checkMemoryPressure:collect:garbage' })
```

File: packages/server/lib/browsers/memory/index.ts [hunk 2]
```diff
@@ -274,10 +275,15 @@ const checkMemoryPressureAndLog = async ({ automation, test }: { automation: Aut
...
       await automation.request('collect:garbage', null, null)
+      span.end()
```
Comment (AtofStryker) [inline packages/server/lib/browsers/memory/index.ts]:
```suggestion
      span?.end()
```

File: packages/server/lib/browsers/memory/index.ts [hunk 3]
```diff
@@ -274,10 +275,15 @@ const checkMemoryPressureAndLog = async ({ automation, test }: { automation: Aut
...
     } catch (err) {
       debug('error collecting garbage: %o', err)
+      span.end()
```
Comment (AtofStryker) [inline packages/server/lib/browsers/memory/index.ts]:
```suggestion
      span?.end()
```

File: packages/server/lib/browsers/memory/index.ts (hunk 1 above)
Comment (mjhenkes) [reply packages/server/lib/browsers/memory/index.ts]: lol, that wouldn't work very well

**Review by chrisbreiding — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

File: packages/server/lib/browsers/memory/index.ts [hunk 4]
```diff
@@ -274,10 +275,15 @@ const checkMemoryPressureAndLog = async ({ automation, test }: { automation: Aut
...
+    let span
+
     try {
+      span = telemetry.startSpan({ name: 'checkMemoryPressure:collect:garbage' })
       await automation.request('collect:garbage', null, null)
+      span?.end()
     } catch (err) {
       debug('error collecting garbage: %o', err)
+      span?.end()
     }
```
Comment (mschile) [inline packages/server/lib/browsers/memory/index.ts]:
```suggestion
    } catch (err) {
      debug('error collecting garbage: %o', err)
    } finally {
      span?.end()
    }
```

**Review by mschile — APPROVED** [review]

**Pull request merged**